src/harness/testcases/WINNF_FT_S_SIQ_testcase.py

```diff
@@ -841,6 +842,231 @@ def test_WINNF_FT_S_SIQ_11(self):
+
+    # 3. Spectrum Inquiry: Missing parameter low frequency and missing cbsdId
+    spectrum_inquiry_3 = json.load(
+        open(os.path.join('testcases', 'testdata', 'spectrum_inquiry_0.json')))
+    spectrum_inquiry_3['inquiredSpectrum'] = [{
+        'highFrequency': 3700000000
+    }]
+    spectrum_inquiry_3['cbsdId'] = "REMOVE"
+
+    # Create the actual config.
+    devices = [device_b, device_c, device_e]
+    conditionals = {
+        'registrationData': [conditionalParameters]
+    }
+
+    config = {
+        'registrationRequestsN2': devices,
+        'conditionalRegistrationData' : conditionals,
+        'expectedResponseCodes': [(0,), (103,), (102,)],  
+        'palRecordsN3':[pal_records1,pal_records2],
+        'ppaRecordsN3':[{'ppaRecord':ppa_record1,
+                         'ppaClusterList': ppa_cluster_list_1},
+                        {'ppaRecord':ppa_record2,
+                         'ppaClusterList': ppa_cluster_list_2}],
+   
+        'spectrumInquiryRequestsN2':[spectrum_inquiry_1,spectrum_inquiry_2,spectrum_inquiry_3],
+        'gwpzRecordsN1':[gwpz_e],
+        'fr1Cbsd': [ [{'frequency' :{'lowFrequency': 3570000000,'highFrequency': 3580000000}} 
+                       ],[],[] ], 
+
+        'fr2Cbsd': [ [ {'frequency' :{'lowFrequency': 3570000000,'highFrequency': 3580000000}}],
+                       [],[] ]
+
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_SIQ_12_default_config)
+  def test_WINNF_FT_S_SIQ_12(self, config_filename):
+    """[Configurable] Spectrum inquiry in the presence of protected entities."""
+
+    config = loadConfig(config_filename)
+
+    # Light checking of the config file
+    self.assertEqual(len(config['spectrumInquiryRequestsN2']),len(config['registrationRequestsN2']))
+    self.assertEqual(len(config['expectedResponseCodes']),len(config['spectrumInquiryRequestsN2']))
+    self.assertEqual(len(config['spectrumInquiryRequestsN2']),len(config['fr1Cbsd']))
+    self.assertEqual(len(config['spectrumInquiryRequestsN2']),len(config['fr2Cbsd']))
+    self.assertGreater(len(config['registrationRequestsN2']),0)
+
+    # Step1: Load information about N1 GWBZs
+    for gwpz_record in config['gwpzRecordsN1']:
+      self._sas_admin.InjectWisp(gwpz_record)
+
+    # Step2&3: Register N2 CBSDs
+    # Check registration response
+    # The assertRegistered function does the Inject FCC ID and user ID for the registration requests
+    cbsd_ids = self.assertRegistered(config['registrationRequestsN2'],config['conditionalRegistrationData'])
+
+    # Update PPA records with devices' CBSD IDs and Inject zone data
+    if ('ppaRecordsN3' in config) and (config['ppaRecordsN3']):
+      for ppa in config['ppaRecordsN3']:
+        if len(ppa['ppaClusterList']) != 0 : 
+          ppa['ppaRecord']['ppaInfo']['cbsdReferenceId'] = []
+        for device_index in ppa['ppaClusterList']:
+          ppa['ppaRecord']['ppaInfo']['cbsdReferenceId'].append(cbsd_ids[device_index])
```

Comment:
Shouldn't this 'for' loop be inside the previous if condition?

Reply:
The   functionality of adding "cbsdReferenceId" would be same even if the   "for" loop is outside the "if" condition as   "for" condition would not be executed if ppa['ppaClusterList'] is   empty.          As suggested, moved "for" loop inside if condition to avoid   confusion